[tensorexpr] Fix isNLC segfault

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72786

Differential Revision: [D34204523](https://our.internmc.facebook.com/intern/diff/D34204523)

IvanKobzarev committed:
[tensorexpr] Fix isNLC segfault

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34204523).

ZolotukhinM reviewed (approved):
Looks good! Would it be possible to add a test where it segfaulted? Also, can we refactor `isNHWC` and `isNLC` to reduce code duplication?

IvanKobzarev committed:
Update on "[tensorexpr] Fix isNLC segfault"


Differential Revision: [D34204523](https://our.internmc.facebook.com/intern/diff/D34204523)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34204523).
